istioctl/cmd/revision.go

```diff
@@ -0,0 +1,1079 @@
+			for i := 0; i < maxIopRows; i++ {
+				var rowTag, rowRev string
+				var rowIopName, rowProfile, rowComp, rowCust string
+				if i == 0 {
+					rowIopName = qualifiedName
+					rowProfile = profile
+				}
+				if i < len(components) {
+					rowComp = components[i]
+				}
+				if i < len(customizations) {
+					rowCust = customizations[i]
+				}
+				if rowId < len(tags) {
+					rowTag = tags[rowId]
+				}
+				if rowId == 0 {
+					rowRev = r
+				}
+				if verbose {
+					fmt.Fprintf(tw, "%s\t%s\t%s\t%s\t%s\t%s\n",
+						rowRev, rowTag, rowIopName, rowProfile, rowComp, rowCust)
+				} else {
+					fmt.Fprintf(tw, "%s\t%s\t%s\t%s\t%s\n",
+						rowRev, rowTag, rowIopName, rowProfile, rowComp)
+				}
+				rowId++
+			}
+		}
+		for rowId < len(tags) {
+			var rowRev, rowTag, rowIopName string
+			if rowId == 0 {
+				rowRev = r
+			}
+			if rowId == 0 {
+				rowIopName = "<no-iop>"
+			}
+			rowTag = tags[rowId]
+			if verbose {
+				fmt.Fprintf(tw, "%s\t%s\t%s\t \t \t \n", rowRev, rowTag, rowIopName)
+			} else {
+				fmt.Fprintf(tw, "%s\t%s\t%s\t \t \n", rowRev, rowTag, rowIopName)
+			}
+			rowId++
+		}
+	}
+	return tw.Flush()
+}
+
+func getAllIstioOperatorCRs(client kube.ExtendedClient) ([]*iopv1alpha1.IstioOperator, error) {
+	ucrs, err := client.Dynamic().Resource(istioOperatorGVR).
+		List(context.Background(), meta_v1.ListOptions{})
+	if err != nil {
+		return []*iopv1alpha1.IstioOperator{}, fmt.Errorf("cannot retrieve IstioOperator CRs: %v", err)
+	}
+	iopCRs := []*iopv1alpha1.IstioOperator{}
+	for _, u := range ucrs.Items {
+		u.SetCreationTimestamp(meta_v1.Time{})
+		u.SetManagedFields([]meta_v1.ManagedFieldsEntry{})
+		iop, err := operator_istio.UnmarshalIstioOperator(util.ToYAML(u.Object), true)
+		if err != nil {
+			return []*iopv1alpha1.IstioOperator{},
+				fmt.Errorf("error while converting to IstioOperator CR - %s/%s: %v",
+					u.GetNamespace(), u.GetName(), err)
+		}
+		iopCRs = append(iopCRs, iop)
+	}
+	return iopCRs, nil
+}
+
+func printRevisionDescription(w io.Writer, args *revisionArgs, logger clog.Logger) error {
+	revision := args.name
+	client, err := newKubeClientWithRevision(kubeconfig, configContext, revision)
+	if err != nil {
+		return fmt.Errorf("cannot create kubeclient for kubeconfig=%s, context=%s: %v",
+			kubeconfig, configContext, err)
+	}
+	allIops, err := getAllIstioOperatorCRs(client)
+	if err != nil {
+		return fmt.Errorf("error while fetching IstioOperator CR: %v", err)
+	}
+	iopsInCluster := getIOPWithRevision(allIops, revision)
+	allWebhooks, err := getWebhooks(context.Background(), client)
+	if err != nil {
+		return fmt.Errorf("error while fetching mutating webhook configurations: %v", err)
+	}
+	webhooks := getWebhooksWithRevision(allWebhooks, revision)
+	revDescription := getBasicRevisionDescription(iopsInCluster, webhooks)
+	if err = enrichWithIOPCustomization(revDescription, args.manifestsPath, logger); err != nil {
+		return err
+	}
+	enrichWithControlPlanePodInfo(revDescription, client)
```

Comment:
suggest s/enrich/annotate/ for these function names

Reply:
Done.